### DISCUSSION TITLE
Export component prop types

### DISCUSSION THREAD

**Discussion by mattoni** [tailwindlabs/headlessui#1213] [Ideas]
Is there any way we could get the prop types for each component exported? I'm attempting to wrap the headless ui components with my own that are styled for my UI library, and ideally I'd like to reuse the props as-is. It seems that they aren't imported and can't be derived using `React.ComponentProps<typeof Dialog>` for example.

Any help is appreciated, thanks!

**Comment by malonecj** [comment]
This would be a big help. Currently, I'm wrapping existing components to provide re-usable styled components and duplicating their props

**Comment by florian-lefebvre** [comment]
+1
